src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/GetPackageReferenceUtility.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using NuGet.ProjectManagement.Projects;
+using System.Threading.Tasks;
+using NuGet.Frameworks;
+using NuGet.LibraryModel;
+using NuGet.Packaging;
+using NuGet.ProjectManagement;
+using NuGet.ProjectModel;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.VisualStudio.Utility
+{
+    internal static class GetPackageReferenceUtility
+    {
+        internal static PackageIdentity UpdateResolvedVersion(LibraryDependency library, NuGetFramework targetFramework, TargetFrameworkInformation assetsTargetFrameworkInformation, IList<LockFileTarget> targets, Dictionary<string, ProjectInstalledPackages> installedPackages)
+        {
+            NuGetVersion resolvedVersion = default;
+
+            if (installedPackages.TryGetValue(library.Name, out ProjectInstalledPackages installedVersion) && installedVersion.AllowedVersions.Equals(library.LibraryRange.VersionRange) && targets == null)
```

Comment:
Given the large amount of parameters, it'll be helpful to either improve the names or add some comments.